src/NeoLocalPlanner.cpp

```diff
@@ -173,7 +173,8 @@ geometry_msgs::msg::TwistStamped NeoLocalPlanner::computeVelocityCommands(
   const geometry_msgs::msg::Twist & speed,
   nav2_core::GoalChecker * goal_checker)
 {
-	boost::mutex::scoped_lock lock(m_odometry_mutex);
+	std::lock_guard<std::mutex> lock_reinit(m_mutex);
```

Comment:
Hm, depends on what you want to achieve. Right now, the effect is that the 3 functions can not be called simultaneously at all, one call blocks the others. I see no problem with deadlocking if that is what you mean.
What exactly are you trying to protect with the mutex? I might need some background about who is calling the member functions and when.

Reply:
> What exactly are you trying to protect with the mutex? 

Alright, there are 2 important components:

1. `m_odometry` should not be updated until the `computeVelocityCommands()` process or generates the necessary velocity commands. 
2.  while a parameter is dynamically changed by the user, it will be updated in / by the `dynamicParametersCallback` if and only if the `computeVelocityCommands()` has finished it's process as in the previous case.  

This is where the confusion arise, if I need to use two mutexes for two different components.. 

whereas in case of the functions being called,  `dynamicParametersCallback()` and `odomCallback()` is being executed by one thread, whereas the `computeVelocityCommands()` is being executed in a different thread. I'm not going into details on where these functions are being called from, it might be too much for you I guess..